### PR TITLE
Add more prose to requests API docs

### PR DESCRIPTION
**Pull request by mgeisler**
In a discussion with Everett Toews, he pointed out that our API docs could use a bit of motivation.

### CODE REVIEW

File: doc/Requests.md [hunk 1]
```diff
@@ -1,83 +1,106 @@
...
+of the data before it's being served.
+
 ## POST
 
-To issue POST request in Zerocloud you need to add `X-Zerovm-Execute: 1.0` header to the request.
+To issue POST request in Zerocloud you need to add
+`X-Zerovm-Execute: 1.0` header to the request.
 
-You can issue several types of POST requests it all depends on what `Content-Type` you pass there.
+This is the most flexible method to start a ZeroVM job. Depending on
```
Comment (etoews) [inline doc/Requests.md:13]: Ah. It wasn't clear to me before that POST was being used to start a ZeroVM job. This is good to know. :)

This really belongs at the top just under the ##POST

File: doc/Requests.md [hunk 2]
```diff
@@ -1,83 +1,106 @@
...
+what `Content-Type` you pass there, the POST payload is interpreted
+differently:
 
 - `Content-Type: application/json` - POST a job description
```
Comment (etoews) [inline doc/Requests.md]: Link "POST a job description" to its section below.

File: doc/Requests.md [hunk 3]
```diff
@@ -1,83 +1,106 @@
...
 
-- `Content-Type: application/x-tar` or `Content-Type: application/x-gtar` or `Content-Type: application/x-ustar` - POST ZeroVM image
+- `Content-Type: application/x-tar` or `Content-Type:
+  application/x-gtar` or `Content-Type: application/x-ustar` - POST a
```
Comment (etoews) [inline doc/Requests.md]: Link "POST a ZeroVM image" to its section below.

File: doc/Requests.md [hunk 4]
```diff
@@ -1,83 +1,106 @@
...
+  ZeroVM image
 
-- any other `Content-Type` - POST a script to be handled by some interpreter or shell executable.
+- any other `Content-Type` - POST a script to be handled by some
```
Comment (etoews) [inline doc/Requests.md]: Link "POST a script" to its section below.

File: doc/Requests.md [hunk 5]
```diff
@@ -1,83 +1,106 @@
...
+  interpreter or shell executable.
 
-## GET
+### POST a job description
+
```
Comment (etoews) [inline doc/Requests.md:28]: What is a job description and why would I want to use it?

A couple of sentences or a link to outside doc would be appropriate here.

File: doc/Requests.md [hunk 6]
```diff
@@ -1,83 +1,106 @@
...
-</pre>
+This is the best use-case for preconfigured jobs where there are
+little changes between the runs. It's best suited for clustered runs,
+where you get a lot of nodes working on different objects and
+communicating with each other.
 
-The supported version strings are:
 
-- open - will execute binary associated with `Content-Type` of the object you're GET-ting, akin to any file manager `double-click` function.
+### POST a ZeroVM image file
```
Comment (etoews) [inline doc/Requests.md:48]: What is a ZeroVM image and why would I want to use it?

A couple of sentences or a link to outside doc would be appropriate here.

File: doc/Requests.md [hunk 7]
```diff
@@ -1,83 +1,106 @@
...
+from the same image file to run on different nodes.
 
 ### POST any other file
 
-This POST will work if url path info is of the form: /version/account or /version/account/container/object
+This POST will work if url path info is of the form:
+`/version/account` or `/version/account/container/object`.
+
+If you issue a POST with `X-Zerovm-Execute: 1.0` and any other content
+type the data will be interpreted as a script.
```
Comment (etoews) [inline doc/Requests.md]: Is the script the body of this request or an existing object in a container?

Comment (pkit) [inline doc/Requests.md]: It's the body of the request

**Comment by pkit** [comment]
We can merge it now or address the suggestions and then merge...

File: doc/Requests.md (hunk 7 above)
Comment (etoews) [inline doc/Requests.md]: So POST is for executing a script that you pass along with the request but GET is for executing a script (or zapp) that is already in a container?

**Comment by etoews** [comment]
Best to address the suggestions before a merge. Easier to track how the suggestions were dealt with that way.

**Comment by mgeisler** [comment]
Either way will work for me — I didn't have time to work on this today.

@everett-toews most of your comments are further enhancements and I think they could be done nicely in a followup pull request. Factual errors in what I wrote should of course be fixed now rather than later.

**Comment by mgeisler** [comment]
@everett-toews thanks for the review, btw! I'll try to make a pass over the text tomorrow and incorporate your suggestions.

File: doc/Requests.md (hunk 7 above)
Comment (pkit) [inline doc/Requests.md]: GET is for historical reasons, described in #72

**Comment by etoews** [comment]
However you want to handle my suggestions is fine. This PR, another one, or not at all. :)

File: doc/Requests.md (hunk 2 above)
Comment (mgeisler) [inline doc/Requests.md]: Good idea, thanks.

File: doc/Requests.md (hunk 3 above)
Comment (mgeisler) [inline doc/Requests.md]: Ok.

File: doc/Requests.md (hunk 4 above)
Comment (mgeisler) [inline doc/Requests.md]: Ok.

File: doc/Requests.md (hunk 5 above)
Comment (mgeisler) [inline doc/Requests.md:28]: I'll rephrase this a little and make a real link to `Servlets.md`.

File: doc/Requests.md (hunk 6 above)
Comment (mgeisler) [inline doc/Requests.md:48]: That is actually described just below. I can rephrase this a little.

**Comment by mgeisler** [comment]
@everett-toews I've tried to address your comments!

@pkit Feel free to merge this now. I suggest we do further followups as another pull request.

**Pull request merged**